common/ledger/snapshot/file.go

```diff
@@ -0,0 +1,200 @@
+	}
+	return nil
+}
+
+// EncodeUVarint encodes and appends a number to the data stream
+func (c *FileCreator) EncodeUVarint(u uint64) error {
+	n := binary.PutUvarint(c.varintReusableBuf, u)
+	if _, err := c.multiWriter.Write(c.varintReusableBuf[:n]); err != nil {
+		return errors.Wrapf(err, "error while writing data to the snapshot file: %s", c.file.Name())
+	}
+	return nil
+}
+
+// Done closes the snapshot file and returns the final hash of the data stream
+func (c *FileCreator) Done() ([]byte, error) {
+	if err := c.bufWriter.Flush(); err != nil {
+		return nil, errors.Wrapf(err, "error while flushing to the snapshot file: %s ", c.file.Name())
+	}
+	if err := c.file.Close(); err != nil {
+		return nil, errors.Wrapf(err, "error while closing the snapshot file: %s ", c.file.Name())
+	}
+	return c.hasher.Sum(nil), nil
+}
+
+// Close closes the underlying file, if not already done. A consumer can invoke this function if the consumer
+// encountered some error and simply wants to abandon the snapshot file creation (typically, intended to be used in a defer statement)
+func (c *FileCreator) Close() error {
+	if c == nil {
+		return nil
+	}
+	return errors.Wrapf(c.file.Close(), "error while closing the snapshot file: %s", c.file.Name())
+}
+
+// FileReader reads from a ledger snapshot file. This is expected to be used for loading the ledger snapshot data
+// during bootstrapping a channel from snapshot. The data should be read, using the functions `DecodeXXX`,
+// in the same sequence in which the data was written by the functions `EncodeXXX` in the `FileCreator`.
+// Note that the FileReader does not verifies the hash of stream and it is expected that the hash has been verified
+// by the consumer. Later, if we decide to perform this, on-the-side, while loading the snapshot data, the FileRedear,
+// like the FileCreator, would take a `hasher` as an input
+type FileReader struct {
+	file              *os.File
+	bufReader         *bufio.Reader
+	reusableByteSlice []byte
+}
+
+// OpenFile constructs a FileReader. This function returns an error if the format of the file, stored in the
+// first byte, does not match with the expectedDataFormat
+func OpenFile(filePath string, expectDataformat byte) (*FileReader, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while opening the snapshot file: %s", filePath)
+	}
+	bufReader := bufio.NewReader(file)
+	dataFormat, err := bufReader.ReadByte()
+	if err != nil {
+		file.Close()
+		return nil, errors.Wrapf(err, "error while reading from the snapshot file: %s", filePath)
+	}
+	if dataFormat != expectDataformat {
+		file.Close()
+		return nil, errors.New(fmt.Sprintf("unexpected data format: %x", dataFormat))
+	}
+	return &FileReader{
+		file:      file,
+		bufReader: bufReader,
+	}, nil
+}
+
+// DecodeString reads and decodes a string
+func (r *FileReader) DecodeString() (string, error) {
```

Comment:
```suggestion
func (r *FileReader) ReadString() (string, error) {
```
Similarly, we can change 

1. `DecodeBytes()` -> `ReadBytes()`
2. `DecodeProtoMessage()` -> `ReadProtoMessage()`
3. `DecodeUVarInt()` -> `ReadUint64()`

Reply:
same as - https://github.com/hyperledger/fabric/pull/1231#discussion_r423054326

Reply:
For file, `ReadXXX()` and `WriteXXX()` is more explicit and natural as compared to `EncodeXXX()` and `DecodeXXX()`.